docs/contributing/index.rst

```diff
@@ -4,200 +4,464 @@
+    to the project without much prior knowledge. If you have already contributed
+    to geomstats, you should look at easy issues instead.
+
+.. topic:: starter project tag
+
+    If you have already contributed to geomstats, another great way to contribute
+    to geomstats is to pick an item from the list of `starter project
+    <https://github.com/geomstats/geomstats/labels/starter%project>`_ in the issue
+    tracker. Your assistance in this area will be greatly appreciated by the
+    more experienced developers as it helps free up their time to concentrate on
+    other issues.
+
+.. topic:: help wanted tag
+
+    We often use the help wanted tag to mark issues regardless of difficulty. Additionally,
+    we use the help wanted tag to mark Pull Requests which have been abandoned
+    by their original contributor and are available for someone to pick up where the original
+    contributor left off. The list of issues with the help wanted tag can be found
+    `here <https://github.com/geomstats/geomstats/labels/help%20wanted>`__ .
+
+    Note that not all issues which need contributors will have this tag.
+
+Making changes
+---------------
+
+The preferred way to contribute to geomstats is to fork the `main
+repository <https://github.com/geomstats/geomstats/>`__ on GitHub,
+then submit a "pull request" (PR).
+
+Follow the setup guidelines detailed in `getting the code`_. In this referenced section, 
+we explain how to locally install geomstats, and how to set up your git repository:
+The next steps now describe the process of modifying code and submitting a PR:
+
+#. Synchronize your master branch with the upstream master branch:
+
+    .. code-block:: bash 
 
         $ git checkout master
         $ git pull upstream master
 
-6. Create a feature branch to hold your development changes::
+#. | Create a feature branch to hold your development changes:
+
+    .. code-block:: bash 
 
         $ git checkout -b my_feature
 
    and start making changes. Always use a feature branch. It's good
    practice to never work on the ``master`` branch!
 
-7. Develop the feature on your feature branch on your computer, using Git to
+#. | Develop the feature on your feature branch on your computer, using Git to
    do the version control. When you're done editing, add changed files using
-   ``git add`` and then ``git commit``::
+   ``git add`` and then ``git commit``:
+
+    .. code-block:: bash 
 
        $ git add modified_files
        $ git commit -m "Add my feature"
 
    to record your changes in Git. Your commit message should respect the `good
    commit messages guidelines <https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project>`_.
-   Then push the changes to your GitHub account with::
+   Then push the changes to your GitHub account with:
+
+    .. code-block:: bash 
 
-       $ git push -u origin my-feature
+         $ git push -u origin my-feature
```

Comment:
maybe do `git push origin <branch-name>` and add a not telling to add flag `-u` if branch does not exist yet remotely.